planner/core/point_get_plan.go

```diff
@@ -901,17 +891,44 @@ func getNameValuePairs(nvPairs []nameValuePair, tblName model.CIStr, expr ast.Ex
 				param = x
 			}
 		} else {
-			return nil
+			return nil, false
 		}
 		if d.IsNull() {
-			return nil
+			return nil, false
+		}
+		// Views' columns have no FieldType.
+		if tbl.IsView() {
+			return nil, false
 		}
 		if colName.Name.Table.L != "" && colName.Name.Table.L != tblName.L {
-			return nil
+			return nil, false
+		}
+		col := model.FindColumnInfo(tbl.Cols(), colName.Name.Name.L)
+		if col == nil || // Handling the case when the column is _tidb_rowid.
+			(col.Tp == mysql.TypeString && col.Collate == charset.CollationBin) { // This type we needn't to pad `\0` in here.
```

Comment:
Any test case for that? I don't know this is used for what.

Reply:
Yes. These tests already exist. The test case for `create table t(a binary(2) primary key, b binary(2));` table, like ` select * from t where a = "a  ";`
@crazycs520